src/data_info/services/stat.clj

```diff
@@ -27,6 +28,24 @@
   [stat-map]
   (= (:permission stat-map) :own))
 
+(defn- needs-key?
+  [included-keys needed-key?]
+  (if (contains? included-keys needed-key?)
+    true
+    (cond
+      (= needed-key? :permission)
+      (contains? included-keys :share-count)
+
+      (= needed-key? :type)
+      (not (empty? (cset/intersection #{:infoType :content-type :file-count :dir-count} included-keys)))
```

Comment:
Yeah, `included-keys` makes sense elsewhere, but something was tripping me up just in this function. Maybe just the `if->case` refactor will help me out more than anything, but `requested-keys` sounds good too 👍